### PR TITLE
Update minimum tested Android version to 8.4

### PR DESCRIPTION
**Pull request by jvandort**
At least one of our deprecation removals breaks compatibility with AGP < 8.4

This updates our testing suite to only test AGP >= 8.4

This includes updating samples to run on modern versions of Android, removing old Android samples which are no longer included in docs, and removing version-specific edge cases from smoke tests

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by jvandort** [comment]
@bot-gradle test this without pts

File: gradle/dependency-management/agp-versions.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # Generated - Update by running `./gradlew updateAgpVersions`
-latests=7.4.2,8.0.2,8.2.2,8.3.2,8.4.2,8.5.2,8.6.1,8.7.3,8.8.1,8.9.0-rc02,8.10.0-alpha05
+latests=8.4.2,8.5.2,8.6.1,8.7.3,8.8.2,8.9.0,8.10.0-beta01
 nightlyBuildId=13097777
 nightlyVersion=8.10.0-dev
```
Comment (jvandort) [inline gradle/dependency-management/agp-versions.properties:4]: Tried to upgrade to `8.11.0-dev`, but there seem to be bugs in the current dev version causing us to run into [this bug](https://youtrack.jetbrains.com/issue/IDEA-369261) in our santa tracker smoke tests. 

So I chose to leave the nightly version as-is

**Comment by jvandort** [comment]
@bot-gradle test ast without pts

**Review by ov7a — APPROVED** [review]
